internal/verifier/migration_verifier.go

```diff
@@ -398,7 +398,7 @@ func (verifier *Verifier) getGenerationWhileLocked() (int, bool) {
 }
 
 func (verifier *Verifier) maybeAppendGlobalFilterToPredicates(predicates bson.A) bson.A {
-	if verifier.globalFilter == nil {
+	if len(verifier.globalFilter) == 0 {
```

Comment:
Why this change?

Reply:
This is to prevent an error like [REP-3390](https://jira.mongodb.org/browse/REP-3390). I don't think it's necessary here since we are only appending the filter to a slice of other predicates. But I think it makes slightly more sense to compare length with 0 than to check for a nil slice.